libs/net/vmspec.py

```diff
@@ -63,15 +90,38 @@ def _lookup_iface_status(vm: BaseVirtualMachine, iface_name: str, predicate: Cal
             the interface needs to meet.
 
     Returns:
-        iface (ResourceField): The requested interface.
-
-    Raises:
-        VMInterfaceStatusNotFoundError: If the requested interface was not found in the VM.
+        iface (ResourceField) | None: The requested interface or None
     """
     for iface in vm.vmi.interfaces:
         if iface.name == iface_name and predicate(iface):
             return iface
-    raise VMInterfaceStatusNotFoundError(f"Network interface named {iface_name} was not found in VM {vm.name}.")
+    return None
+
+
+def wait_for_iface_status_deleted(vm: BaseVirtualMachine, interface_name: str, wait_timeout: int = 30) -> None:
```

Comment:
The name implies as if you know who called this function. The name should reflect the operation intent with no knowledge about why it was called (e.g. it may be because one unplugged the interface and it may be because one wants to confirm the specific interface is not there in the first place).

Alternative names:
- `wait_for_missing_iface_status`
- `confirm_missing_iface_status`